server/src/core/server/services/users/emailAliasBanFilter.ts

```diff
@@ -0,0 +1,36 @@
+import { MongoContext } from "coral-server/data/context";
+import { parseEmailAliasIntoParts, sanitizeStringForRegex } from "./helpers";
+
+export const shouldBanEmailBecauseOtherAliasesAreBanned = async (
+  mongo: MongoContext,
+  email?: string | null
+) => {
+  if (!email) {
+    return false;
+  }
+
+  const { isAlias, baseEmail: base } = parseEmailAliasIntoParts(email);
+  if (!isAlias || !base) {
+    return false;
+  }
+
+  const regexFriendlyAddress = sanitizeStringForRegex(
+    base.baseEmailWithoutDomain
+  );
+  const regexFriendlyDomain = sanitizeStringForRegex(base.domain);
+  const regex = new RegExp(`^${regexFriendlyAddress}.*${regexFriendlyDomain}$`);
+
+  const usersCursor = mongo.users().find({ email: { $regex: regex } });
+  while (await usersCursor.hasNext()) {
+    const user = await usersCursor.next();
+    if (!user) {
+      continue;
+    }
+
+    if (user.status.ban.active) {
+      return true;
```

Comment:
I noticed that in the user drawer account history tab in the admin, these bans are described as `Account domain banned.` I think that this was simply the only way that accounts were banned by the system up until now. We might want to either update this text (`moderate-user-drawer-account-history-account-domain-banned`) to clarify that it could be because the account was domain banned or banned for being an alias of a banned email. Or else figure out a way to determine which reason it was banned for.

Reply:
ty!